packages/services/src/__tests__/transactionEvent.test.ts

```diff
@@ -0,0 +1,150 @@
+    identifierKeyPaths: ['transaction', 'identifier'],
+  };
+
+  it('should process transaction event', async () => {
+    (vckitService.issueVC as jest.Mock).mockImplementationOnce(() => transactionVCMock);
+    (uploadJson as jest.Mock).mockImplementationOnce(() => uploadedTransactionEventLinkMock);
+    jest
+      .spyOn(validateContext, 'validateTransactionEventContext')
+      .mockReturnValueOnce({ ok: true, value: context } as Result<ITransactionEventContext>);
+    jest.spyOn(helpers, 'getIdentifierByObjectKeyPaths').mockReturnValueOnce(nlisidMock);
+    jest.spyOn(linkResolverService, 'registerLinkResolver').mockResolvedValueOnce(transactionEventDLRMock);
+
+    const transactionVC = await processTransactionEvent(transactionEvent, context);
+
+    expect(transactionVC).toBe(transactionVCMock);
+    expect(uploadJson).toHaveBeenCalled();
+    expect(validateContext.validateTransactionEventContext).toHaveBeenCalled();
+    expect(helpers.getIdentifierByObjectKeyPaths).toHaveBeenCalled();
+    expect(linkResolverService.registerLinkResolver).toHaveBeenCalled();
+  });
+
+  it('should throw error if context validation throws an error', async () => {
+    try {
+      jest
+        .spyOn(validateContext, 'validateTransactionEventContext')
+        .mockReturnValueOnce({ ok: false, value: 'Invalid context' });
+
+      await processTransactionEvent(transactionEvent, context);
+    } catch (e) {
+      const error = e as Error;
+      expect(error.message).toBe('Invalid context');
+    }
+  });
+
+  it('should throw error if identifier not found', async () => {
+    try {
+      jest
+        .spyOn(validateContext, 'validateTransactionEventContext')
+        .mockReturnValueOnce({ ok: true, value: context } as Result<ITransactionEventContext>);
+      jest.spyOn(helpers, 'getIdentifierByObjectKeyPaths').mockReturnValueOnce(undefined);
+
+      await processTransactionEvent(transactionEvent, context);
+    } catch (e) {
+      const error = e as Error;
+      expect(error.message).toBe('Identifier not found');
+      expect(validateContext.validateTransactionEventContext).toHaveBeenCalled();
+    }
+  });
+
+  it('should throw error if issueVC throws an error', async () => {
+    try {
+      jest
+        .spyOn(validateContext, 'validateTransactionEventContext')
+        .mockReturnValueOnce({ ok: true, value: context } as Result<ITransactionEventContext>);
+      jest.spyOn(helpers, 'getIdentifierByObjectKeyPaths').mockReturnValueOnce(nlisidMock);
+      jest.spyOn(publicAPI, 'post').mockRejectedValueOnce("Can't issue VC");
+
+      await processTransactionEvent(transactionEvent, context);
+    } catch (e) {
+      const error = e as Error;
+      expect(error.message).toBe("Can't issue VC");
+      expect(validateContext.validateTransactionEventContext).toHaveBeenCalled();
+      expect(helpers.getIdentifierByObjectKeyPaths).toHaveBeenCalled();
+    }
```

Comment:
the `processTransactionEvent` does not handle try-catch when error throw for the calling services

Reply:
I think we should keep the original error thrown by the `processTransactionEvent` function. The developer implementing that function will catch those errors and handle them at their discretion.